joomla/joomla-cms
# [4.0] Better error handling in com_finder/sample data

Pull Request for Issue #26036 (not a patch but better error handling).

### Summary of Changes
Adds better error handling in two place

1. The plugin itself - now if non-json responses come back due to PHP errors then the plugin will now display a pretty notice and stop showing loading forever
2. Taxonomy. Something is very wrong in finder @Hackwar . Seems very related to https://github.com/joomla/joomla-cms/pull/26235 - stuff in taxonomy tables seems either missing or incomplete - only one category out of them all was showing in the table itself.

### Testing Instructions
Install sample data

1. Optionally comment out the new code in the taxonomy table. Install sample data, see the error message that the response is invalid (note the english in that message is not great - but is an existing string so is outside scope of this PR - i'm sure @brianteeman will patch that)
2. (if step 1 was performed commenting out the finder error handling then undo that) Install sample data - see error message about nodes not existing in the smart search taxonomy table.

### Expected result
Error handling not causing sample data plugins to stall

### Actual result
Error handling is in place both for the actual issue we currently have in finder and also for any future things in the modules javascript

### Documentation Changes Required
None
